apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -76,11 +73,25 @@ function MyTokens({
 
 function TokenDetails({ coinType }: TokenDetailsProps) {
     const { loading, error, showError } = useObjectsState();
-    const activeCoinType = coinType || GAS_TYPE_ARG;
+    const activeCoinType = coinType || SUI_TYPE_ARG;
     const accountAddress = useAppSelector(({ account }) => account.address);
-    const balances = useAppSelector(accountAggregateBalancesSelector);
-    const tokenBalance = balances[activeCoinType] || BigInt(0);
-    const allCoinTypes = useMemo(() => Object.keys(balances), [balances]);
+    const { data: coinBalance, isLoading: loadingBalances } = useGetAllBalance({
+        address: accountAddress,
+    });
+
+    const tokenBalance = useMemo(() => {
```

Comment:
Why not have this use the individual fetching route?

Reply:
Done